QRCoder/QRCodeGenerator.Polynom.cs

```diff
@@ -8,21 +9,131 @@ public partial class QRCodeGenerator
         /// <summary>
         /// Represents a polynomial, which is a sum of polynomial terms.
         /// </summary>
-        private struct Polynom
+        private struct Polynom : IDisposable
         {
+            private PolynomItem[] _polyItems;
+            private int _length;
+
             /// <summary>
             /// Initializes a new instance of the <see cref="Polynom"/> struct with a specified number of initial capacity for polynomial terms.
             /// </summary>
             /// <param name="count">The initial capacity of the polynomial items list.</param>
             public Polynom(int count)
             {
-                this.PolyItems = new List<PolynomItem>(count);
+                _length = 0;
+                _polyItems = Allocate(count);
+            }
+
+            /// <summary>
+            /// Adds a polynomial term to the polynomial.
+            /// </summary>
+            public void Add(PolynomItem item)
+            {
+                EnsureCapacity(_length + 1);
+                _polyItems[_length++] = item;
+            }
+
+            /// <summary>
+            /// Removes the polynomial term at the specified index.
+            /// </summary>
+            public void RemoveAt(int index)
+            {
+                if (index < 0 || index >= _length)
+                    throw new IndexOutOfRangeException();
+
+                if (index < _length - 1)
+                    Array.Copy(_polyItems, index + 1, _polyItems, index, _length - index - 1);
+
+                _length--;
+            }
+
+            /// <summary>
+            /// Gets or sets a polynomial term at the specified index.
+            /// </summary>
+            public PolynomItem this[int index]
+            {
+                get {
+                    if (index < 0 || index >= _length)
+                        throw new IndexOutOfRangeException();
+                    return _polyItems[index];
+                }
+                set {
+                    if (index < 0 || index >= _length)
+                        throw new IndexOutOfRangeException();
+                    _polyItems[index] = value;
```

Comment:
W/ the `uint`-trick from above the bound-checks can be elided by the JIT here too.

Reply:
Done